src/Config/ConfigResolver.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * Slim Framework (https://slimframework.com)
+ *
+ * @license https://github.com/slimphp/Slim-Console/blob/0.x/LICENSE.md (MIT License)
+ */
+
+declare(strict_types=1);
+
+namespace Slim\Console\Config;
+
+use InvalidArgumentException;
+use RuntimeException;
+use Slim\Console\Config\Parser\JSONConfigParser;
+use Slim\Console\Config\Parser\PHPConfigParser;
+use Slim\Console\Exception\CannotResolveConfigException;
+
+class ConfigResolver
+{
+    public const CONFIG_FILENAME = 'slim-console.config';
+
+    public const FORMAT_PHP = 'php';
+
+    public const FORMAT_JSON = 'json';
+
+    /**
+     * @var string[]
+     */
+    protected $supportedFormats = [
+        self::FORMAT_PHP,
+        self::FORMAT_JSON,
+    ];
+
+    /**
+     * @var string
+     */
+    protected $rootDir;
+
+    /**
+     * @param string $rootDir
+     */
+    public function __construct(string $rootDir)
+    {
+        $this->rootDir = $rootDir;
+    }
+
+    /**
+     * Resolve configuration. Environment takes precedence over configuration file.
+     *
+     * @return Config
+     *
+     * @throws CannotResolveConfigException
+     * @throws InvalidArgumentException
+     * @throws RuntimeException
+     */
+    public function resolve(): Config
+    {
+        try {
+            return Config::fromEnvironment();
+        } catch (InvalidArgumentException $e) {
+            return $this->attemptResolvingConfigFromSupportedFormats();
+        }
+    }
+
+    /**
+     * @return Config
+     *
+     * @throws CannotResolveConfigException
+     * @throws InvalidArgumentException
+     * @throws RuntimeException
+     */
+    protected function attemptResolvingConfigFromSupportedFormats(): Config
+    {
+        foreach ($this->supportedFormats as $format) {
+            $path = $this->rootDir . DIRECTORY_SEPARATOR . self::CONFIG_FILENAME . ".{$format}";
```

Comment:
Micro-Optimization: Instead of concatenating in every loop, we could also write
```php
        $basePath = $this->rootDir . DIRECTORY_SEPARATOR . self::CONFIG_FILENAME . '.';
        foreach ($this->supportedFormats as $format) {
            $path = $basePath . $format;
```
But maybe, that is less readable (maintainable).

Reply:
I think this is fine, the method is pretty small so this micro-optimization is still readable. Done!